File: pyg_manager_web/src/main/webapp/js/service/specificationService.js
//只和后台交互 不和页面进行交互  不要出现this
app.service("specificationService", function ($http) {
//分页查询
    this.findPage = function (pageNo, pageSize) {
        return $http.get("../specification/findPage?pageNo=" + pageNo + "&pageSize=" + pageSize);
    };

    //查询所有
    this.findAll = function () {
        return $http.get("../specification/findAll");
    };

    //保存
    //把保存方法拆开
    this.add = function (entity) {
        return $http.post("../specification/add", entity);
    };
    this.update = function (entity) {
        return $http.post("../specification/update", entity);
    };


    //根据id查询对象
    this.findOne = function (id) {
        return $http.get("../specification/findOne/" + id);
    };
    this.dele = function (selectedIds) {

        return $http.get("../specification/dele/" + selectedIds);
    };

    this.searchEntity = {};
    //条件查询
    this.search = function (pageNo, pageSize, searchEntity) {
        //当前页码,每页显示条数,查询的对象searchEntity
        return $http.post("../specification/search?pageNo=" + pageNo + "&pageSize=" + pageSize, searchEntity);
    };
});

File: pyg_sellergoods_interface/src/main/java/com/pyg/sellergoods/service/SpecificationService.java
package com.pyg.sellergoods.service;

import com.pyg.pojo.TbSpecification;
import entity.PageResult;
import groupEntity.Specification;

import java.util.List;
import java.util.Map;

/**
 * @ClassName SpecificationService
 * @Description TODO
 * @Author Deric
 * @Date 2019/1/20 19:17
 * @Version 1.0
 **/
public interface SpecificationService {

    public List<TbSpecification> findAll();

    PageResult findPage(int pageNo, int pageSize);

    void add(Specification specification);

    Specification findOne(Long id);

    public void update(Specification specification);

    void dele(Long[] ids);

//    public List<TbSpecification> search(TbSpecification specification);

    public PageResult findPage(int pageNo, int pageSize, TbSpecification specification);

    List<Map> findSpecList();

}
